google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -60,6 +60,18 @@ void GetBucketMetadata(storage::Client client, int& argc, char* argv[]) {
 }
 //! [get bucket metadata]
 
+//! [list objects]
+void ListObjects(storage::Client client, int& argc, char* argv[]) {
+  if (argc < 2) {
+    throw Usage{"list-objects <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  for (auto const& meta : client.ListObjects(bucket_name)) {
+    std::cout << "gs://" << meta.bucket() << "/" << meta.name() << std::endl;
```

Comment:
Outside of gsutil, we shouldn't encourage using the gs:// style of URL for identifying objects.  It ends up leading to weird naming convention confusion for users.

Reply:
Fixed.